lib/nationbuilder/client.rb

```diff
@@ -16,6 +16,10 @@ def initialize(nation_name, api_key, opts = {})
     end
   end
 
+  def http_client
```

Comment:
Apologies for the delay on reviewing this -

Exposing this publicly to modify the underlying `HTTPClient` after an instance of `Nationbuilder::Client` has been instantiated doesn't feel like the best order to allow additional configuration options for the former. 

In thinking of another way to solve this, the `NationBuilder::Client` takes an `opts` hash - that feels like an appropriate point to allow a (configured) instance of `HTTPClient` to be passed in while still defaulting to new one otherwise, something like:

```ruby
def initialize(nation_name, api_key, opts = {})
  ...
  @http_client = opts[:http_client] || HTTPClient.new
end
```

Thoughts? 

cc: @nationbuilder/internal-collaborators